api/v1/logicalvolume_types.go

```diff
@@ -18,6 +18,22 @@ type LogicalVolumeSpec struct {
 	NodeName    string            `json:"nodeName"`
 	Size        resource.Quantity `json:"size"`
 	DeviceClass string            `json:"deviceClass,omitempty"`
+
+	// 'type' specifies the logical volume type.
+	// Can be 'thin-snapshot' or left empty in case of a regular volume.
+	// +kubebuilder:validation:Optional
+	Type string `json:"type"`
+
+	// 'sourceID' specifies the volumeID of the parent logical volume; if present.
```

Comment:
```suggestion
	// 'sourceID' specifies the parent LogicalVolume name; if present.
```

We talked about storing the LogicalVolume name instead of the volumeID in the sourceID. Are there plans to change this in the future?

Reply:
@llamerada-jp  yes, it holds the LogicalVolume name from the backend: 
```
[root@fedora bin]# lvs
  LV                                   VG    Attr       LSize Pool     Origin                               Data%  Meta%  Move Log Cpy%Sync Convert
  4f0cfd20-e70d-4d88-9642-3a6db713fb7f myvg1 Vri-a-tz-- 2.00g thinpool cff35deb-5b6c-4194-806c-fa9481ecc62f 0.52                                                                    
  cff35deb-5b6c-4194-806c-fa9481ecc62f myvg1 Vwi-aotz-- 2.00g thinpool                                      0.52            
```
```
[root@fedora bin]# ./kubectl get LogicalVolume snapshot-3f1f336d-db18-4f88-b570-b767dc308b25 -oyaml
apiVersion: topolvm.cybozu.com/v1
kind: LogicalVolume
metadata:
  creationTimestamp: "2022-04-26T09:34:43Z"
  finalizers:
  - topolvm.cybozu.com/logicalvolume
  generation: 1
  labels:
    app.kubernetes.io/created-by: topolvm-controller
  name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25
  resourceVersion: "76079"
  uid: 4f0cfd20-e70d-4d88-9642-3a6db713fb7f
spec:
  deviceClass: ssd-thin
  name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25
  nodeName: topolvm-example-worker
  size: 2Gi
  snapshot:
    accessType: ro
    sourceID: cff35deb-5b6c-4194-806c-fa9481ecc62f
    type: thin
status:
  currentSize: 2Gi
  volumeID: 4f0cfd20-e70d-4d88-9642-3a6db713fb7f
```

Reply:
(PS: ignore the old spec it is changed now)

Reply:
Sorry for the lack of clarity. I will write `LogicalVolume CR name` next time.

This is a normal volume and is the source volume of snapshot.
```
apiVersion: topolvm.cybozu.com/v1
kind: LogicalVolume
metadata:
  name: pvc-f32b6803-3364-4689-9f8b-45c5415cec52 <- This one.
spec:
  name: pvc-f32b6803-3364-4689-9f8b-45c5415cec52
  nodeName: 10.20.30.40
  size: 100Gi
  deviceClass ssd-thin
status:
  currentSize: 100Gi
  volumeID: 3e178dbf-2f9a-4de2-9c7b-3a45cfdba943 <- Not this one.
```

This is a snapshot volume. The  `LogicalVolume CR name` is set on `sourceID`.
```
apiVersion: topolvm.cybozu.com/v1
kind: LogicalVolume
metadata:
  name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25
spec:
  name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25
  nodeName: 10.20.30.40
  size: 100Gi
  deviceClass ssd-thin
  sourceID: pvc-f32b6803-3364-4689-9f8b-45c5415cec52
  type: thin-snapshot
```

Reply:
Hi @llamerada-jp @nbalacha , 
Regarding https://github.com/topolvm/topolvm/pull/463/files#r860340050 

I believe in place of replacing sourceVolID with source Volume name, 
It will be better if we add both the information on the Logical Volume CR.

Some benefits of the same:
1) It will provide more information for debugging, as we will have both origin LogicalVolume CR name along with its origin's volumeID.
2) Since we already have `volumeID` as a part of our LogicalVolume CR;
keeping the sourceID in case of `snapshots`, `restored-volumes` and `clones` will give us a better understanding of from which volume this is created.

Here is an example:
```
apiVersion: topolvm.cybozu.com/v1 
kind: LogicalVolume 
metadata: 
	name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25 
spec: 
	name: snapshot-3f1f336d-db18-4f88-b570-b767dc308b25
	source: pvc-f32b6803-3364-4689-9f8b-45c5415cec52
 	nodeName: 10.20.30.40 
	size: 100Gi 
	deviceClass ssd-thin 
	sourceID: pvc-f32b6803-3364-4689-9f8b-45c5415cec52 
	type: thin-snapshot
```

What do you think?

Reply:
Hi, @Yuggupta27 
I suggest you only have the name of LogicalVolume CR. The reasons are as follows.
- If having the name of LogicalVolume CR and LV volume id, we need to pay attention to those consistencies.
- The name of LogicalVolume CR is useful for implementing the remote copy features in the future. We can't determine whether the source volume is remote or local using only the LV volume id.
- According to KEP, the origin PV/PVC is not deleted while the snapshot exists. LogicalVolume CR is also present if they are not deleted manually and irregularly.
- The origin can be deleted even if the copy exists in the clone feature. However, that is the same even if using LV volume id. At present, there is no problem because the origin LogicalVolume will not be referred to the copy source LogicalVolume CR after the data copy is completed.